.gitignore

```diff
@@ -21,3 +21,7 @@ build
 # matlab / octave
 *.mat
 octave-workspace
+
+# clion
+.idea/
+cmake-build-*/
```

Comment:
Are these changes required for this PR? Are they related? I expect we want a universal .gitignore that includes them, but we want to avoid changing them to whatever people want to use locally on random PRs. Don't we have a standard for .gitignore?